src/Mvc/Mvc.ViewFeatures/src/Rendering/SystemTextJsonHelper.cs

```diff
@@ -10,20 +11,47 @@ namespace Microsoft.AspNetCore.Mvc.Rendering
 {
     internal class SystemTextJsonHelper : IJsonHelper
     {
-        private readonly JsonOptions _options;
+        private readonly JsonSerializerOptions _htmlSafeJsonSerializerOptions;
 
         public SystemTextJsonHelper(IOptions<JsonOptions> options)
         {
-            _options = options.Value;
+            _htmlSafeJsonSerializerOptions = CreateHtmlSafeSerializerOptions(options.Value.JsonSerializerOptions);
         }
 
         /// <inheritdoc />
         public IHtmlContent Serialize(object value)
         {
             // JsonSerializer always encodes non-ASCII chars, so we do not need
             // to do anything special with the SerializerOptions
-            var json = JsonSerializer.Serialize(value, _options.JsonSerializerOptions);
+            var json = JsonSerializer.Serialize(value, _htmlSafeJsonSerializerOptions);
             return new HtmlString(json);
         }
+
+        private static JsonSerializerOptions CreateHtmlSafeSerializerOptions(JsonSerializerOptions serializerOptions)
+        {
+            var htmlSafeOptions = new JsonSerializerOptions
```

Comment:
@ahsonkhan @steveharter  - these are the kinds of cases where a copy constructor would be massively useful.

Reply:
+1 as a future feature.

Reply:
I see. Being tracked by an issue that @pranavkm opened earlier: https://github.com/dotnet/corefx/issues/39939